WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController+DomainCredit.swift

```diff
@@ -18,10 +18,15 @@ extension BlogDetailsViewController {
     }
 
     @objc func showDomainCreditRedemption() {
-        // TODO-#11467 - subtask - integration with domain registration.
-        // Temporarily shows success screen before domain registration is integrated.
-        let domain = "lifeoftea.com"
-        presentDomainCreditRedemptionSuccess(domain: domain)
+        guard let site = JetpackSiteRef(blog: blog) else {
+            return
```

Comment:
Is it worth logging an error here, or this case should not happen and `fatalError` is more suitable?

Reply:
@pinarol thanks for reviewing it. I like more the second solution. Displaying an alert and log the error might be the best solution avoiding to add another case to handle the Domain Credit Cell logic.
```
I couldn't think of a valid case where JetpackSiteRef returns nil and at the same time canRedeemDomainCredit returns true.
```
I agree with this. But if it might happen then I prefer log the error and display the alert.

Reply:
Just updated with temporary copy in https://github.com/wordpress-mobile/WordPress-iOS/pull/11613/commits/a54f173335b8e7f938b1306e7101690e2a69e167, will update after the copy is confirmed.

In the same commit, also added code to deselect currently selected row so that "Register domain" row doesn't stay selected after error alert is dismissed. Deselecting currently selected row was not needed before for row with action that presents a modal, because `tableView.reloadData()` is called in `viewWillAppear` that is triggered on fullscreen modal dismissal. Dismissing an alert does not trigger `viewWillAppear`.